webapp/views.py

```diff
@@ -87,6 +90,22 @@ def releasenotes_redirect():
         return flask.redirect(f"https://wiki.ubuntu.com/Releases")
 
 
+def search_snaps():
+    """
+    A JSON endpoint to search the snap store API
+    """
+
+    query = flask.request.args.get("q", "")
+    arch = flask.request.args.get("arch", "amd64")
+    size = flask.request.args.get("size", "100")
+    page = flask.request.args.get("size", "1")
+
+    if not query:
+        flask.abort(400)
+
+    return store_api.search(query, size=size, page=page, arch=arch)
```

Comment:
I think you can wrap this in `flask.jsonify` to have the correct format of response, the mimetype and all:
https://flask.palletsprojects.com/en/1.1.x/api/#flask.json.jsonify

(if store_api.search returns just a dictionary.)